packages/x-components/src/x-modules/search-box/components/search-input-placeholder.vue

```diff
@@ -0,0 +1,241 @@
+<template>
+  <component
+    :is="animation"
+    v-if="isVisible"
+    class="x-input-placeholder x-search-input-placeholder"
+    mode="out-in"
+  >
+    <span v-if="isBeingAnimated" :key="animationMessage" data-test="search-input-placeholder">
+      {{ animationMessage }}
+    </span>
+    <span v-else-if="$slots.default" data-test="search-input-placeholder">
+      <!--
+        @slot The default placeholder when the `messages` animation is not active.
+      -->
+      <slot />
+    </span>
+  </component>
+</template>
+
+<script lang="ts">
+  import Vue from 'vue';
+  import { Component, Prop } from 'vue-property-decorator';
+
+  import { animateTranslate, State, xComponentMixin, XOn } from '../../../components';
+  import { searchBoxXModule } from '../x-module';
+
+  /**.
+   * This component renders an animated placeholder for the search input in the shape of a list of
+   * iterating messages that can be configured to happen always or only when hovering the input
+   *
+   * @public
+   */
+  @Component({
+    mixins: [xComponentMixin(searchBoxXModule)]
+  })
+  export default class SearchInputPlaceholder extends Vue {
+    /**
+     * List of messages to animate.
+     */
+    @Prop({ required: true })
+    protected messages!: Array<string>;
+
+    /**
+     * Animation component used for the messages.
+     *
+     * @public
+     */
+    @Prop({ default: () => animateTranslate('bottom-to-top') })
+    protected animation!: Vue;
+
+    /**
+     * Time in milliseconds during which each message is visible.
+     *
+     * @public
+     */
+    @Prop({ default: 1500 })
+    protected animationIntervalMs!: number;
+
+    /**
+     * Whether the messages animation is active only when hovering the search input or always.
+     *
+     * @public
+     */
+    @Prop({ default: false })
+    protected animateOnlyOnHover!: boolean;
+
+    @State('searchBox', 'query')
+    public query!: string;
+
+    protected isSearchBoxHovered = false;
+
+    protected isSearchBoxFocused = false;
+
+    protected animationMessageIndex = 0;
+
+    protected animationInterval: number | undefined;
+
+    protected get isVisible(): boolean {
+      return !this.query && !this.isSearchBoxFocused;
```

Comment:
actually, the usual behaviour of an `<input>` placeholder is to show it even it has focus, and only hide when there is something typed in the input:

https://user-images.githubusercontent.com/5759712/195584978-a633f8cb-6f46-4bf4-acfb-48a7654006bc.mov

Reply:
Talked offline about keeping it as it is because of the issue's requirements, it can be easily changed in the future if the approach mentioned here is seen more convenient.